pkg/snapshotter/snapshotter_csi.go

```diff
@@ -640,6 +655,22 @@ func (c *csiDriver) RestoreVolumeClaim(opts ...Option) (*v1.PersistentVolumeClai
 			}
 
 		}
+		// PB-8482 if restore size is nil then set the pvc size to annotation restoreSize
+		if snapshot.Status.RestoreSize == nil {
+			restoreSizeStr, ok := snapshot.Annotations["px-backup/restoreSize"]
+			if !ok {
+				return nil, fmt.Errorf("restoreSize not found either in status or in annotation")
+			}
+			restoreSize, err := resource.ParseQuantity(restoreSizeStr)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse restoreSize quantity provided in annotation px-backup/restoreSize: %v", err)
```

Comment:
nit: in continuation with above comment. Such hard coded values become difficult manage in future.